.tests/alexa.rb

```diff
@@ -1,86 +1,96 @@
+timestamp = Time.now.utc.strftime('%Y%m%dT%H%M%SZ')
+datestamp = Time.now.utc.strftime('%Y%m%d')
 
 headers = {
-  "host"        => SERVICE_ENDPOINT,
-  "x-amz-date"  => timestamp
+    'host' => SERVICE_ENDPOINT,
+    'x-amz-date' => timestamp
 }
 
 query = {
-  "Action"           => action,
-  "ResponseGroup"    => responseGroup,
-  "Url"              => site
+    Action: action,
+    ResponseGroup: responsegroup,
+    Url: site
 }
 
-query_str = query.sort.map{|k,v| k + "=" + escapeRFC3986(v.to_s())}.join('&')
-headers_str = headers.sort.map{|k,v| k + ":" + v}.join("\n") + "\n"
-headers_lst = headers.sort.map{|k,v| k}.join(";")
-payload_hash = Digest::SHA256.hexdigest ""
-canonical_request = "GET" + "\n" + SERVICE_URI + "\n" + query_str + "\n" + headers_str + "\n" + headers_lst + "\n" + payload_hash
-algorithm = "AWS4-HMAC-SHA256"
-credential_scope = datestamp + "/" + SERVICE_REGION + "/" + SERVICE_NAME + "/" + "aws4_request"
-string_to_sign = algorithm + "\n" +  timestamp + "\n" +  credential_scope + "\n" + (Digest::SHA256.hexdigest canonical_request)
-signing_key = getSignatureKey(secret_access_key, datestamp, SERVICE_REGION, SERVICE_NAME)
-signature=OpenSSL::HMAC.hexdigest('sha256', signing_key, string_to_sign)
-authorization_header = algorithm + " " + "Credential=" + access_key_id + "/" + credential_scope + ", " +  "SignedHeaders=" + headers_lst + ", " + "Signature=" + signature;
-
-url = "https://" + SERVICE_HOST + SERVICE_URI + "?" + query_str
+query_str = query.sort.map { |k, v| "#{k}=#{escaperfc3986(v.to_s)}" }.join('&')
+headers_str = headers.sort.map { |k, v| "#{k}:#{v}" }.join("\n") + "\n"
+headers_lst = headers.sort.map { |k, _v| k }.join(';')
+payload_hash = Digest::SHA256.hexdigest ''
+canonical_request = "GET\n#{SERVICE_URI}\n#{query_str}\n#{headers_str}\n#{headers_lst}\n#{payload_hash}"
+algorithm = 'AWS4-HMAC-SHA256'
+credential_scope = "#{datestamp}/#{SERVICE_REGION}/#{SERVICE_NAME}/aws4_request"
+string_to_sign = "#{algorithm}\n#{timestamp}\n#{credential_scope}\n#{(Digest::SHA256.hexdigest canonical_request)}"
+signing_key = getsignaturekey(secret_access_key, datestamp, SERVICE_REGION, SERVICE_NAME)
+signature = OpenSSL::HMAC.hexdigest('sha256', signing_key, string_to_sign)
+authorization_header = "#{algorithm} Credential=#{access_key_id}/#{credential_scope}, SignedHeaders=#{headers_lst}, Signature=#{signature}"
+
+url = "https://#{SERVICE_HOST}#{SERVICE_URI}?#{query_str}"
 uri = URI(url)
 req = Net::HTTP::Get.new(uri)
-req["Accept"] = "application/xml"
-req["Content-Type"] = "application/xml"
-req["x-amz-date"] = timestamp
-req["Authorization"] = authorization_header
+req['Accept'] = 'application/xml'
+req['Content-Type'] = 'application/xml'
+req['x-amz-date'] = timestamp
+req['Authorization'] = authorization_header
 
 res = Net::HTTP.start(uri.host, uri.port,
-  :use_ssl => uri.scheme == 'https') {|http|
+                      use_ssl: uri.scheme == 'https') do |http|
   http.request(req)
-}
+end
```

Comment:
Lines 72-83 is a bit strange cos it passes in the URI twice. I think it can be combined into:
```suggestion
uri = URI("https://#{SERVICE_HOST}#{SERVICE_URI}?#{query_str}")
request_headers = {
  'Accept' => 'application/xml',
  'Content-Type' => 'application/xml',
  'x-amz-date' => timestamp,
  'Authorization' => authorization_header
}

res = Net::HTTP.get_response(uri, request_headers)
```